### PR TITLE
[runtime env] move worker env to runtime env in Java

### PR DESCRIPTION
**Pull request by SongGuyang**
## Why are these changes needed?
Remove the field "worker_env" in job config and use runtime_env.
Use `setup_worker.py` to setup Java workers.

## Related issue number

#18213

### CODE REVIEW

**Review by wuisawesome — APPROVED** [review]
proto changes look backwards compatible

**Comment by edoakes** [comment]
Oops, @SongGuyang could you mark me and @architkulkarni as "assigned" when you need a review? Our reviewing workflow filters to PRs we're assigned to.

File: python/ray/_private/runtime_env/context.py [hunk 1]
```diff
@@ -34,10 +34,13 @@ def serialize(self) -> str:
     def deserialize(json_string):
         return RuntimeEnvContext(**json.loads(json_string))
 
-    def exec_worker(self, passthrough_args: List[str]):
+    def exec_worker(self, passthrough_args: List[str], language: str):
```
Comment (edoakes) [inline python/ray/_private/runtime_env/context.py:37]: Can we make language an ENUM please?

File: src/ray/core_worker/core_worker.cc [hunk 2]
```diff
@@ -31,30 +31,6 @@ namespace core {
...
-      current_task_id, task_index, caller_id, address, num_returns, required_resources,
-      required_placement_resources, bundle_id, placement_group_capture_child_tasks,
-      debugger_breakpoint, serialized_runtime_env, runtime_env_uris,
-      concurrency_group_name);
-  // Set task arguments.
-  for (const auto &arg : args) {
-    builder.AddArg(*arg);
-  }
-}
-
```
Comment (edoakes) [inline src/ray/core_worker/core_worker.cc:57]: what's the reason for moving this?

File: src/ray/core_worker/core_worker.h [hunk 3]
```diff
@@ -1051,6 +1051,17 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
...
+      TaskSpecBuilder &builder, const JobID &job_id, const TaskID &task_id,
+      const std::string &name, const TaskID &current_task_id, const uint64_t task_index,
+      const TaskID &caller_id, const rpc::Address &address, const RayFunction &function,
+      const std::vector<std::unique_ptr<TaskArg>> &args, uint64_t num_returns,
+      const std::unordered_map<std::string, double> &required_resources,
+      const std::unordered_map<std::string, double> &required_placement_resources,
+      const BundleID &bundle_id, bool placement_group_capture_child_tasks,
+      const std::string &debugger_breakpoint, const std::string &serialized_runtime_env,
+      const std::vector<std::string> &runtime_env_uris,
+      const std::string &concurrency_group_name = "");
```
Comment (edoakes) [inline src/ray/core_worker/core_worker.h:1064]: hmm I'm confused wasn't this already here? did you make it private? if so, remove the public one?

File: src/ray/raylet/worker_pool.cc [hunk 4]
```diff
@@ -313,37 +313,43 @@ Process WorkerPool::StartWorkerProcess(
...
+      }
+
+      worker_command_args.push_back("--runtime-env-hash=" +
+                                    std::to_string(runtime_env_hash));
+
+      if (serialized_runtime_env_context != "{}" &&
+          serialized_runtime_env_context != "") {
+        worker_command_args.push_back("--serialized-runtime-env-context=" +
+                                      serialized_runtime_env_context);
+      }
```
Comment (edoakes) [inline src/ray/raylet/worker_pool.cc:335]: are there any changes here or just re-ordered?

**Review by edoakes — COMMENTED** [review]
Some clarifying questions -- it looks like some things maybe just got moved around?

File: src/ray/raylet/worker_pool.cc (hunk 4 above)
Comment (SongGuyang) [reply src/ray/raylet/worker_pool.cc:335]: > Oops, @SongGuyang could you mark me and @architkulkarni as "assigned" when you need a review? Our reviewing workflow filters to PRs we're assigned to.

OK. Sounds good.

File: src/ray/core_worker/core_worker.h (hunk 3 above)
Comment (SongGuyang) [reply src/ray/core_worker/core_worker.h:1064]: I moved this function from a static function(only defined in "core_worker.cc") to a member function of `CoreWorker` class. This function is only used by `CoreWorker`. And the original reason is that I need to use `job_config_` to write runtime_env in this function. `job_config_` is a member of `CoreWorker`.

Comment (edoakes) [reply src/ray/core_worker/core_worker.h:1064]: Gotcha, makes sense!

File: python/ray/_private/runtime_env/context.py (hunk 1 above)
Comment (SongGuyang) [reply python/ray/_private/runtime_env/context.py]: Good suggestion. I have used the language enum from protobuf. https://github.com/ray-project/ray/pull/19060/commits/3341139cef661d959606071ac8b9606697bb4156

**Review by edoakes — APPROVED** [review]
LGTM once tests pass

**Review by architkulkarni — APPROVED** [review]

**Pull request merged**